### PR TITLE
[BEAM-3740][Draft][WIP][Prototype] Retractions in DirectRunner

### PR DESCRIPTION
**Pull request by akedin**
Tracking the progress in this PR

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand:
   - [ ] What the pull request does
   - [ ] Why it does it
   - [ ] How it does it
   - [ ] Why this approach
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by akedin** [comment]
@kennknowles @apilloud  FYI

File: model/pipeline/src/main/resources/org/apache/beam/model/common_urns.md [hunk 1]
```diff
@@ -125,6 +125,8 @@ but are used at runtime and must be supported by every SDK.
 
 ###  beam:coder:windowed_value:v1
 
+###  beam:coder:windowed_value:v2
```
Comment (lukecwik) [inline model/pipeline/src/main/resources/org/apache/beam/model/common_urns.md]: @jkff had worked on getting the URNs into an enum. Best to sync with him on that.

Comment (akedin) [reply model/pipeline/src/main/resources/org/apache/beam/model/common_urns.md]: Ah, good to know, will do when this is out of exploratory stage. With this version I mostly want to clarify the scope, APIs, and unknowns. This code may eventually get thrown away, depending on findings.

Comment (kennknowles) [reply model/pipeline/src/main/resources/org/apache/beam/model/common_urns.md]: If it is a [spike](https://en.wikipedia.org/wiki/Spike_(software_development)) in the classic sense, then I do enjoy being able to see the code in a PR, and interested people who don't sit next to Anton can also take a peek.

**Pull request closed without merging**